control/state.py

```diff
@@ -24,19 +24,22 @@ class GatewayState(ABC):
         X_PREFIX: Key prefix for key of type "X"
     """
 
-    BDEV_PREFIX = "bdev_"
     NAMESPACE_PREFIX = "namespace_"
     SUBSYSTEM_PREFIX = "subsystem_"
     HOST_PREFIX = "host_"
     LISTENER_PREFIX = "listener_"
-
-    def build_bdev_key(bdev_name: str) -> str:
-        return GatewayState.BDEV_PREFIX + bdev_name
+    NAMESPACE_QOS_PREFIX = "qos_"
```

Comment:
would it make sense to store namespace qos definition under corresponding namespace?

Reply:
We don't get these values when we create the namespace, only in the set_qos command.

Reply:
we could store qos under corresponding namespace.

Reply:
Already answered that. I wanted to maintain the existing usage of OMAP. We save the request with the OMAP resource. The namespace create request doesn't have the QOS limits. We only have them in the set_qos request. I didn't see a value in complicating the code just to avoid using an extra key type in OMAP.

Reply:
Gave it some thought. I could have added the QOS settings to the namespace_add request and then set them after creating the namespace. But, this would only cover the case of creating a namespace. If we want to use OMAP to update the QOS settings for an existing namespace we need to have a separate object for them. Adding them to the namespace object means the code would need to understand what was changed in the namespace and take different actions accordingly. Either deleting and creating the namespace or just call a qos_set function. Unless you want to recreate the namespace for any QOS change. Again, I think the price of adding a new key type to OMAP is much smaller than the alternative.